src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,108 @@
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number;
+};
+
+function verifyUniqueCommentId(
+  existingComments: Comment[],
+  newComment: Comment,
+): boolean {
+  const existingIds = existingComments.map(({ id }) => id);
+
+  return !existingIds.includes(newComment.id);
+}
+
+export const PostDetails: React.FC<Props> = ({ selectedPostId }) => {
+  const [postDetails, setPostDetails] = useState<Post | null>(null);
+  const [postComments, setPostComments] = useState<Comment[] | []>([]);
+  const [areCommentsVisible, setCommentsVisibility] = useState<boolean>(true);
+
+  useEffect(() => {
+    const loadData = async () => Promise.all([
+      getPostDetails(selectedPostId),
+      getPostComments(selectedPostId),
+    ]);
+
+    loadData().then(([post, comments]) => {
+      setPostDetails(post);
+      setPostComments(comments);
+    });
+  }, [selectedPostId]);
+
+  const deleteComment = useCallback((commentId: number) => {
+    const updatedComments = postComments.filter(({ id }) => id !== commentId);
+
+    setPostComments(updatedComments);
+  }, [postComments]);
+
+  const addNewComment = useCallback((newComment: Comment) => {
+    if (!verifyUniqueCommentId(postComments, newComment)) {
+      const updatedComment = {
+        ...newComment,
+        id: newComment.id + 1,
+      };
+
+      addNewComment(updatedComment);
+    } else {
+      setPostComments([newComment, ...postComments]);
+    }
+  }, [postComments]);
+
+  return postDetails && (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{postDetails.body}</p>
+      </section>
+
+      {postComments.length && (
+        <>
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              className="button"
+              onClick={() => setCommentsVisibility(!areCommentsVisible)}
+            >
+              {areCommentsVisible ? 'Hide' : 'Show'}
+              &nbsp;
+              {postComments.length}
+              {' '}
+              comments
```

Comment:
temlate string

Reply:
linter does not want to keep it in the single line